setup/tue-functions.bash

```diff
@@ -295,34 +283,29 @@ function tue-revert
 {
     human_time="$*"
 
-    fs=`ls $_TUE_CATKIN_SYSTEM_DIR/src`
-    for pkg in $fs
+    for pkg_dir in $_TUE_CATKIN_SYSTEM_DIR/src/*/
     do
-        pkg_dir=$_TUE_CATKIN_SYSTEM_DIR/src/$pkg
+        pkg=$(basename $pkg_dir)
 
-        if [ -d $pkg_dir ]
+        branch=$(git -C $pkg_dir rev-parse --abbrev-ref HEAD 2>&1)
+        if [ $? -eq 0 ] && [ $branch != "HEAD" ]
         then
-            cd $pkg_dir
-            branch=$(git rev-parse --abbrev-ref HEAD 2>&1)
-            if [ $? -eq 0 ] && [ $branch != "HEAD" ]
+            new_hash=$(git -C $pkg_dir  rev-list -1 --before="$human_time" $branch)
+            current_hash=$(git -C $pkg_dir  rev-parse HEAD)
+            git -C $pkg_dir  diff -s --exit-code $new_hash $current_hash
+            if [ $? -eq 0 ]
             then
-                new_hash=$(git rev-list -1 --before="$human_time" $branch)
-                current_hash=$(git rev-parse HEAD)
-                git diff -s --exit-code $new_hash $current_hash
-                if [ $? -eq 0 ]
-                then
-                    newtime=$(git show -s --format=%ci)
-                    printf "\e[0;36m%-20s\033[0m %-15s \e[1m%s\033[0m %s\n" "$branch is fine" "$new_hash" "$newtime" "$pkg"
-                else
-                    git checkout -q $new_hash
-                    newbranch=$(git rev-parse --abbrev-ref HEAD 2>&1)
-                    newtime=$(git show -s --format=%ci)
-                    echo $branch > .do_not_commit_this
-                    printf "\e[0;36m%-20s\033[0m %-15s \e[1m%s\033[0m %s\n" "$newbranch based on $branch" "$new_hash" "$newtime" "$pkg"
-                fi
+                newtime=$(git -C $pkg_dir  show -s --format=%ci)
+                printf "\e[0;36m%-20s\033[0m %-15s \e[1m%s\033[0m %s\n" "$branch is fine" "$new_hash" "$newtime" "$pkg"
             else
-                echo "Package $pkg could not be reverted, current state: $branch"
+                git -C $pkg_dir  checkout -q $new_hash
+                newbranch=$(git -C $pkg_dir  rev-parse --abbrev-ref HEAD 2>&1)
+                newtime=$(git -C $pkg_dir  show -s --format=%ci)
+                echo $branch > .do_not_commit_this
```

Comment:
Because you aren't switching directories anymore. This file could be any, because you can run this command at any time in any directory.